sources/api/thar-be-settings/src/service.rs

```diff
@@ -1,46 +1,94 @@
+use crate::{error, Result};
+use itertools::join;
 use snafu::{ensure, OptionExt, ResultExt};
-use std::collections::HashMap;
-use std::collections::HashSet;
+use std::collections::{HashMap, HashSet};
 use std::path::Path;
-use std::process;
-
-use itertools::join;
+use std::process::Command;
+
+// TODO: thar-be-settings isn't used as a library; declare its modules in main rather than lib so
+// we don't have to expose helper types like this just so we can call related functions in main.
+/// The `Services` type is an augmented version of `model::Services` that also stores the list of
+/// settings that were changed and that are relevant to each service.
+#[derive(Debug, Default)]
+pub struct Services(pub HashMap<String, Service>);
+/// The `Service` type stores the original `model::Service` and the list of settings that have
+/// changed that are relevant to that service.
+#[derive(Debug)]
+pub struct Service {
+    /// The specific list of settings that changed and are relevant to this service.  Will be None
+    /// if the program is running for *all* services, like at startup.
+    pub changed_settings: Option<HashSet<String>>,
+    /// The model's representation of this service.
+    pub model: model::Service,
+}
 
-use crate::{error, Result};
+impl Services {
+    /// Convert from the model's representation of all services to our own, adding in the lists of
+    /// changed settings, if appropriate; pass None if the program is running for *all* services.
+    pub fn from_model_services(
+        input: model::Services,
+        mut changed_settings: Option<HashMap<String, HashSet<String>>>,
+    ) -> Self {
+        let mut output = HashMap::new();
+        for (name, model) in input {
+            let changed_settings = changed_settings.as_mut().and_then(|s| s.remove(&name));
```

Comment:
Changing the name of this inner variable to anything other than `changed_settings` would help. I found the function confusing because I didn't quite get the nuance of what was happening with variable shadowing and scope.

```suggestion
            let current_changed_settings = changed_settings.as_mut().and_then(|s| s.remove(&name));
```

Reply:
The field in the service structure is `changed_settings`, and I think that makes sense since they're the changed settings relevant to that service, and if this variable name matches the field then the code is slightly simpler...  An alternate that I think achieves the same goal would be to rename the function parameter from `changed_settings` to `all_changed_settings` or something similar to indicate that it includes all changed settings we know about for all services, which we need to scope down here...

Reply:
OK, yeah that makes more sense.

Reply:
I'll rename the function parameter then